### ISSUE TITLE
[bug]: `MintAsset` returns the entire `MintingBatch`, not just the currently minted `PendingAsset`

### ISSUE THREAD

**Issue by ZZiigguurraatt** [lightninglabs/taproot-assets#1201]
The response of `MintAsset` (https://lightning.engineering/api-docs/api/taproot-assets/mint/mint-asset/index.html) is a `MintAssetResponse` object that contains one field, `pending_batch` which contains the entire `MintingBatch` and not just the currently minted `PendingAsset`. Therefor, we have no idea what asset we just minted if there are more than one `PendingAsset` and the pending assets have the same `name`. Also, we get in the response all pending assets from the batch this newly minted asset is a part of, which is a bit confusing and does not seem like it is data we should expect when minting an asset (I'm not sure why other assets in the batch are relevant to be included in the response when we never asked for them).

I think `MintAsset` should only return a `PendingAsset` object for the currently minted asset. Also, I think the `PendingAsset` object should include a `batch_key` so that we can then find the `MintingBatch` this asset is part of using the `ListBatches` RPC (https://lightning.engineering/api-docs/api/taproot-assets/mint/list-batches/index.html), _if_ we want to. Using `ListBatches` is a more intentional way to request information on pending batches than just automatically sending as part of a response to `MintAsset`.

**Comment by gijswijs** [comment]
The latest version of `tapd` does not allow to have pending assets with the same name in the `MintingBatch`. It will return an error stating `[tapcli] unable to mint asset: rpc error: code = Unknown desc = unable to mint asset: asset with name beefbux already in batch`

We could still choose to only return a `PendingAsset` but I wouldn't classify this (anymore) as a bug.

**Comment by ZZiigguurraatt** [comment]
> The latest version of `tapd` does not allow to have pending assets with the same name in the `MintingBatch`. It will return an error stating `[tapcli] unable to mint asset: rpc error: code = Unknown desc = unable to mint asset: asset with name beefbux already in batch`
> 
> We could still choose to only return a `PendingAsset` but I wouldn't classify this (anymore) as a bug.

Yes, I've confirmed this is the case. Not sure when this changed or if I missed something, but if we can only have a single asset with the same name in the current batch, I think this can be worked around.
